source/stdx/collections/rcstring.d

```diff
@@ -0,0 +1,244 @@
+///
+module stdx.collections.rcstring;
+
+import stdx.collections.common;
+import stdx.collections.array;
+
+debug(CollectionRCString) import std.stdio;
+
+version(unittest)
+{
+    import std.experimental.allocator.mallocator;
+    import std.experimental.allocator.building_blocks.stats_collector;
+    import std.experimental.allocator : RCIAllocator, RCISharedAllocator,
+           allocatorObject, sharedAllocatorObject;
+    import std.algorithm.mutation : move;
+
+    private alias SCAlloc = StatsCollector!(Mallocator, Options.bytesUsed);
+}
+
+///
+struct RCString
+{
+private:
+    Array!ubyte _support;
+    AllocatorHandler _allocator;
+
+    /// Constructs the ouroboros allocator from allocator if the ouroboros
+    // allocator wasn't previously set
+    /*@nogc*/ nothrow pure @safe
+    bool setAllocator(A)(ref A allocator)
+    if (is(A == RCIAllocator) || is(A == RCISharedAllocator))
+    {
+        if (_allocator.isNull)
+        {
+            auto a = typeof(_allocator)(allocator);
+            move(a, _allocator);
+            return true;
+        }
+        return false;
+    }
```

Comment:
All this code is the same in the collections, so maybe use mixins to avoid code duplication and make further changes easier?
After all, supporting shared collections is a goal too, isn't it?